fetch/api/response/response-fragment.window.js

```diff
@@ -0,0 +1,13 @@
+promise_test(() => {
+  return fetch("./#test").then(res => {
+    assert_equals(res.url, new URL("./", location).href);
+    return fetch("../resources/redirect.py?location=/#test").then(res2 => {
+      // redirect.py changes the URL a bit; the important bit here is that the fragment is dropped
+      assert_equals(res2.url, new URL("/?location=%2F&count=1", location).href);
+      // redirect.py ends up appending some stuff to the fragment, but that does not affect this
+      return fetch("../resources/redirect.py?location=/%23test").then(res3 => {
+        assert_equals(res3.url, new URL("/", location).href);
```

Comment:
I was surprised here, but I guess the deal is that fragments on redirects are not preserved and if they're included they're thrown away, *except* for navigations.

Reply:
That's the current deal, yes. Perhaps we should alter the deal though.